### ISSUE TITLE
parse error for trailing comma in tsconfig.json

### ISSUE THREAD

**Issue by dominikg** [TypeStrong/tsconfig#30]
According to this issue on typescript repo, trailing commas in tsconfig are legal (like comments)
https://github.com/microsoft/TypeScript/issues/20384#issuecomment-348552936

Example:
```js
{
  "extends": "@tsconfig/svelte/tsconfig.json",
  "include": ["src/**/*"],
  "exclude": ["node_modules"], // <- here
}
```
Comments are stripped here, but the comma is not, resulting in an error.
https://github.com/TypeStrong/tsconfig/blob/3d0586d30a85e1098d4a966e6b563d58abc42620/src/tsconfig.ts#L187

**Comment by blakeembrey** [comment]
Makes sense, related to https://github.com/TypeStrong/tsconfig/issues/28. Happy to accept a PR here, but would likely mean trying to implement a custom JSON parser. I'd recommend using typescript directly where possible.

**Comment by dominikg** [comment]
I see 2 possible implementations
1) after stripping comments with https://github.com/sindresorhus/strip-json-comments use a similar implementation to strip the dangling commas before passing it to JSON.parse
2) get nasty and use eval, trying to safeguard against nefarious input eg
```js
function parseObjectLiteral(x) {
    // remove anything outside of {}
    const trimmed = x.substring(x.indexOf('{'),x.lastIndexOf('}')+1);
    // escape possibly nefarious things that are not escaped already (not immediately preceded by uneven number of \)
    // this leads to a parser error if they happen to be outside a string literal
    const escaped = trimmed.replace(/(?<!\\(?:[\\]{2})*)([().;=`']|self|window|global|globalThis|this|prototype|constructor|Function|Array|Object|Number|String)/g,'\\$1');
    // set use strict and bind to empty 
    return new Function(`"use strict";return ({x:${escaped}})`).bind(Object.create(null))().x;
}
```

but this could still be prone to regex attacks and i might have missed something that needs escaping. Plus it's more lenient than typescripts own  pseudo"json" format

an optional peerdependency on typescript and using its native functions before resorting to a custom implementation would be better. In my case i ran into this in vite, which uses esbuild so there is no typescript by default.

**Comment by tmeindle** [comment]
Have you thought about using the jsonc-parser package to read the config files ?
https://www.npmjs.com/package/jsonc-parser

**Comment by DecimalTurn** [comment]
@blakeembrey did you ever get the chance to look at @dominikg's original PR (https://github.com/TypeStrong/tsconfig/pull/31)?
If you think it has a good approach, I could always replicate it.

**Comment by dominikg** [comment]
as mentioned in that PR i am no longer interested in this. please do not tag me
